PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ["res.company"]
-    _inherit = ['res.config.settings']
-    _inherit = ["res.users"]

Total Changes: 7

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/auth_ldap/models/res_company.py
+++ b/addons/auth_ldap/models/res_company.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = ["res.company"]
+    _inherit = "res.company"
 
     ldaps = fields.One2many('res.company.ldap', 'company', string='LDAP Parameters',
                                copy=True, groups="base.group_system")

--- a/addons/auth_ldap/models/res_company_ldap.py
+++ b/addons/auth_ldap/models/res_company_ldap.py
@@ -13,6 +13,7 @@ _logger = logging.getLogger(__name__)
 
 
 class ResCompanyLdap(models.Model):
+    _name = 'res.company.ldap'
     _description = 'Company LDAP configuration'
     _order = 'sequence'
     _rec_name = 'ldap_server'

--- a/addons/auth_ldap/models/res_config_settings.py
+++ b/addons/auth_ldap/models/res_config_settings.py
@@ -5,6 +5,6 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     ldaps = fields.One2many(related='company_id.ldaps', string="LDAP Parameters", readonly=False)

--- a/addons/auth_ldap/models/res_users.py
+++ b/addons/auth_ldap/models/res_users.py
@@ -8,7 +8,7 @@ from odoo.modules.registry import Registry
 
 
 class ResUsers(models.Model):
-    _inherit = ["res.users"]
+    _inherit = "res.users"
 
     def _login(self, credential, user_agent_env):
         try:
